plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/internal/CloudSdkPreferenceResolver.java

```diff
@@ -39,8 +47,21 @@ public CloudSdkPreferenceResolver() {
 
   @Override
   public Path getCloudSdkPath() {
+    // We only consult the Managed Cloud SDK when it has been explicitly configured, which
+    // is done in CloudSdkPreferences.
+    if (!preferences.isDefault(CloudSdkPreferences.CLOUD_SDK_MANAGEMENT)) {
```

Comment:
Shouldn't this be `preferences.getString(CLOUD_SDK_MANAGEMENT).isEmpty()`? I think `isDefault()` is basically `.getString(CLOUD_SDK_MANAGEMENT).equals("AUTOMATIC")` here.

Perhaps we don't need this check at all, since we will return the managed Cloud SDK path only when `CloudSdkPreferences.CLOUD_SDK_MANAGEMENT` is set to `"AUTOMATIC"` by the following line, and if it is set to `AUTOMATIC`, it means the preferences are completely set up.

Reply:
Hmm… the javadoc for `isDefault()` is a bit ambiguous:
```java
    /**
     * Returns whether the current value of the preference with the given name
     * has the default value.
     *
     * @param name the name of the preference
     * @return <code>true</code> if the preference has a known default value
     * and its current value is the same, and <code>false</code> otherwise
     * (including the case where the preference is unknown to this store)
     */
    public boolean isDefault(String name);
```
The `@return` description's "_the preference has a known default value and its current value is the same_" can be interpreted as either _the set value is the same as the default value_ or _the value returned is taken from the default value_.

 In practice, `ScopedPreferenceStore` and JFace's `PreferenceStore` use the latter interpretation: that the value would be taken from the defaults.

In reality that's the only useful interpretation as it's impossible to do a check like what I'm doing here: if we only have a default then …